### DISCUSSION TITLE
Does not build with Flutter 2.10

### DISCUSSION THREAD

**Discussion by masewo** [singerdmx/flutter-quill#640] [General]
After upgrading from Flutter 2.8 to Flutter 2.10 my app is not building anymore.

This is the build error log when trying to build for Android:

```
/C:/Users/masewo/AppData/Roaming/Pub/Cache/hosted/pub.dartlang.org/flutter_quill-3.9.4/lib/src/widgets/raw_editor.dart:235:9: Error: Type 'TextEditingActionTarget' not found.
        TextEditingActionTarget,
        ^^^^^^^^^^^^^^^^^^^^^^^
/C:/Users/masewo/AppData/Roaming/Pub/Cache/hosted/pub.dartlang.org/flutter_quill-3.9.4/lib/src/widgets/raw_editor.dart:230:7: Error: The type 'TextEditingActionTarget' can't be mixed in.
class RawEditorState extends EditorState
      ^
/C:/Users/masewo/AppData/Roaming/Pub/Cache/hosted/pub.dartlang.org/flutter_quill-3.9.4/lib/src/widgets/raw_editor.dart:888:11: Error: Superclass has no method named 'copySelection'.
    super.copySelection(cause);
          ^^^^^^^^^^^^^
/C:/Users/masewo/AppData/Roaming/Pub/Cache/hosted/pub.dartlang.org/flutter_quill-3.9.4/lib/src/widgets/raw_editor.dart:914:11: Error: Superclass has no method named 'cutSelection'.
    super.cutSelection(cause);
          ^^^^^^^^^^^^
/C:/Users/masewo/AppData/Roaming/Pub/Cache/hosted/pub.dartlang.org/flutter_quill-3.9.4/lib/src/widgets/raw_editor.dart:939:11: Error: Superclass has no method named 'pasteText'.
    super.pasteText(cause); // ignore: unawaited_futures
          ^^^^^^^^^
/C:/Users/masewo/AppData/Roaming/Pub/Cache/hosted/pub.dartlang.org/flutter_quill-3.9.4/lib/src/widgets/raw_editor.dart:950:11: Error: Superclass has no method named 'selectAll'.
    super.selectAll(cause);
          ^^^^^^^^^#
```

**Comment by masewo** [comment]
We can also discuss here about the consequences of this build error but for me it's actually an issue at the moment.
Made a quick and dirty fix in my fork though: https://github.com/masewo/flutter-quill/commit/0334af6ab5cbe71a3247baf1a8956ee3428a1f79

**Reply by singerdmx** [reply]
If you think it is a good fix, please submit a PR.
Or could you let me know what is your concern for this `dirty fix`?

**Reply by singerdmx** [reply]
@masewo Your dirty fix does not seem to work

**Reply by masewo** [reply]
@singerdmx You are right. Copy and Cut actions are only working for 1 and 2 lines of text.
It just copied over the old implementation of `TextEditingActionTarget` to be able to build my app.
Your solution in https://github.com/singerdmx/flutter-quill/pull/646 seems to work pretty fine for me.

**Reply by singerdmx** [reply]
Hmm it is working for me

**Comment by singerdmx** [comment]
Still WIP but anyone who is interested can give it a try https://github.com/singerdmx/flutter-quill/pull/646
